manifests/java17.pp

```diff
@@ -0,0 +1,28 @@
+#
+# @summary Install jdk 17 via RPMs
+#
+# @param rpm
+#   Name of the rpm to install.
+#
+# @param package
+#   String naming the package installed by the rpm; eg zulu-17.
+#
+class java_artisanal::java17 (
+  String $rpm = 'zulu17.x86_64.rpm',
+  String $package = 'zulu-17',
+) {
+  $rpmfile = "/var/tmp/${rpm}"
+
+  archive { $rpmfile:
+    ensure   => present,
+    source   => "${java_artisanal::pkgurl}/${rpm}",
+    username => $java_artisanal::pkgurl_user,
+    password => $java_artisanal::pkgurl_pass,
+  }
+
+  package { $package:
+    ensure   => 'latest',
```

Comment:
This is probably not the `ensure` value you want went fetching directly from a URL.  I'm not sure what it will do... it might try to download the RPM on every agent run.

Reply:
The package resource is actually installing from a local rpm, that is downloaded by the archive statement.
I thought that would give me the desired behaviour: if/when "zulu17.rpm" in nexus is replaced by a new version, it will get installed.

Reply:
In that case, it probably comes down to if the version / epoch is changing in the metadata in the RPM.